src/app.js

```diff
@@ -72,46 +81,45 @@ export default class App extends React.Component {
     return paginatedListings
   }
   handleSubmit(requestData) {
-    this.setState({view: 'load'})
+    location.hash = 'load'
```

Comment:
Instead of dedicating a route to `load`, just have an `isLoading` state that your `uploader` screen uses to display the spinner until it's done. There isn't really a good reason to add a load screen entry to the browser's history.